tox.ini

```diff
@@ -59,20 +59,38 @@ commands =
     python --version
     pip-audit {env:EXCLUDES:}
 
-[testenv:docs,docs-py{310,311,312}]
+[testenv:docs]
 description = Build documentation, including the examples, and check links
-extras = docs
+skip_install = True
 setenv =
     SMOKE_TEST = true
+    DOCS_LOCKFILE_PATH = .lockfiles/py310-dev.lock
 commands = 
     python --version
+    uv pip sync {env:DOCS_LOCKFILE_PATH}
+    uv pip install .
     python docs/scripts/build_documentation.py {posargs}
 
-[testenv:docs,docs-quickbuild-py{310,311,312}]
+[testenv:docs-quickbuild]
 description = Force-build documentation, ignoring links and examples
-extras = docs
+skip_install = True
 setenv =
     SMOKE_TEST = true
+    DOCS_LOCKFILE_PATH = .lockfiles/py310-dev.lock
 commands = 
     python --version
-    python docs/scripts/build_documentation.py -f
\ No newline at end of file
+    uv pip sync {env:DOCS_LOCKFILE_PATH}
+    uv pip install .
+    python docs/scripts/build_documentation.py -f
+
+[testenv:docs-lockfile]
```

Comment:
When should this be called? Manually whenever we observe that the check fails?

Reply:
Manually when the CI reminds you to do so